src/main/java/pl/edu/pw/mini/ingreedio/api/user/service/UserService.java

```diff
@@ -158,8 +159,9 @@ public boolean unlikeProduct(Integer userId, Long productId) {
         return true;
     }
 
+
     @Transactional
-    public void deleteProduct(Long productId) {
+    public void allUsersUnlikeProduct(Long productId) {
         List<User> users = userRepository.findUsersByLikedProduct(productId);
 
         for (User user : users) {
```

Comment:
this function should not exist, we should do that by a stored procedure:
https://www.postgresql.org/docs/current/xproc.html
https://www.postgresql.org/docs/current/sql-createprocedure.html

which can then be called by
```java
@Procedure("DELETE_PRODUCT")
int deleteProduct(long id);
```
which has to be put in some Repository

Reply:
Why? Because it is very wasteful to load all users who like that product (for example 10k) just to update them in the DBMS. This wastes time and resources.

Reply:
thanks for the resources, I will add it to the todo list for the final refactoring